cmd/common/oci.go

```diff
@@ -0,0 +1,140 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"context"
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"github.com/inspektor-gadget/inspektor-gadget/cmd/common/frontends/console"
+	"github.com/inspektor-gadget/inspektor-gadget/cmd/common/utils"
+	gadgetcontext "github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-context"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-service/api"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/logger"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/params"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/runtime"
+)
+
+func NewRunCommand(rootCmd *cobra.Command, runtime runtime.Runtime, hiddenColumnTags []string) *cobra.Command {
+	runtimeGlobalParams := runtime.GlobalParamDescs().ToParams()
+	runtime.Init(runtimeGlobalParams)
+
+	// Add operator global flags
+	operatorsGlobalParamsCollection := operators.GlobalParamsCollection()
+
+	cmd := &cobra.Command{
+		Use:          "oci",
+		Short:        "run image based gadget",
+		SilenceUsage: true, // do not print usage when there is an error
+		// We have to disable flag parsing in here to be able to handle certain
+		// flags more dynamically and have `--help` also react to those changes.
+		// Instead, we need to manually
+		// * call cmd.ParseFlags()
+		// * handle `--help` after changing the params dynamically
+		// * handle everything that could have been handled inside
+		//   `PersistentPreRun(E)` of a parent cmd, as the flags wouldn't have
+		//   been parsed there (e.g. --verbose)
+		DisableFlagParsing: true,
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			err := runtime.Init(runtimeGlobalParams)
+			if err != nil {
+				return fmt.Errorf("initializing runtime: %w", err)
+			}
+			defer runtime.Close()
+
+			// we need to re-enable flag parsing, as utils.ParseEarlyFlags() would
+			// not do anything otherwise
+			cmd.DisableFlagParsing = false
+
+			// Parse flags that are known at this time, like the ones we get from the gadget descriptor
+			if err := utils.ParseEarlyFlags(cmd, args); err != nil {
+				return err
+			}
+
+			// gadget parameters that are only available after contacting the server
+			gadgetParams := make(params.Params, 0)
+
+			// Before running the gadget, we need to get the gadget info to perform
+			// different tasks like creating the parser and setting flags for the
+			// gadget's parameters.
+			actualArgs := cmd.Flags().Args()
+
+			gadgetCtx := gadgetcontext.NewSimple(context.Background(), actualArgs[0], logger.DefaultLogger())
+
+			// Fetch gadget information; TODO: this can potentially be cached
+			info, err := runtime.GetOCIGadgetInfo(gadgetCtx, nil, args)
+			if err != nil {
+				return fmt.Errorf("fetching gadget information: %w", err)
+			}
+			log.Debugf("GI %+v %v", info, err)
```

Comment:
err is nil at this point.
```suggestion
			log.Debugf("GadgetInfo %+v", info)
```